src/array/cpu/sddmm.cc

```diff
@@ -41,9 +41,22 @@ namespace aten {
     }                                                                   \
   } while (0)
 
+#define SWITCH_BITS(bits, DType, ...)                           \
+  do {                                                          \
+    if ((bits) == 16 || (bits) == 32) {                         \
+      typedef float DType;                                      \
```

Comment:
Do we have comments or documentation saying the CPU FP16 is done with FP32?

Reply:
No and I'll introduce this behavior in FAQ if you think this PR is ready to be merged.